lib/que/leaky_bucket.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Que
+  # Provide a mechanism to throttle resource usage, by specifying a target budget of
+  # resource time to be consumed over a rolling window.
+  #
+  # This is not thread-safe, and should be used by only one thread at any one time.
+  class LeakyBucket
+    def initialize(window:, budget:)
+      @window = window
+      @budget = budget
+      @ratio = budget / window
+      @remaining = 0.0
+      @last_refill = nil
+    end
+
+    # Observe resource usage from within the provided block. This consumes time that
+    # remains in the bucket, and should only be called after the bucket has been refilled.
+    def observe
```

Comment:
Maybe there's something I'm missing about the Prometheus client or something but I can't see where this gets called 🤔